cli/ballerina-cli-utils/src/main/ballerina/packaging_pull/packaging_pull.bal

```diff
@@ -1,72 +1,143 @@
+        logFormatter = new BuildLogFormatter();
+    }
+
+    // resolve endpoint
+    http:Client httpEndpoint;
+    if (host != "" && port != "") {
+        try {
+            httpEndpoint = defineEndpointWithProxy(url, host, port, proxyUsername, proxyPassword);
+        } catch (error err) {
+            return createError("failed to resolve host : " + host + " with port " + port);
+        }
+    } else  if (host != "" || port != "") {
+        return createError("both host and port should be provided to enable proxy");
+    } else {
+        httpEndpoint = defineEndpointWithoutProxy(url);
+    }
+
+    return pullPackage(httpEndpoint, url, pkgPath, dirPath, versionRange, fileSeparator, terminalWidth);
+}
+
+# Pulling a package
+#
+# + httpEndpoint - The endpoint to call
+# + url - Central URL
+# + pkgPath - Package Path
+# + dirPath - Directory path
+# + versionRange - Balo version range
+# + fileSeparator - System file separator
 # + terminalWidth - Width of the terminal
-# + versionRange - Supported version range
-function pullPackage (http:Client definedEndpoint, string url, string dirPath, string pkgPath, string fileSeparator,
-                      string terminalWidth, string versionRange) {
-    endpoint http:Client httpEndpoint = definedEndpoint;
+# + return - Error if occurred
+function pullPackage(http:Client httpEndpoint, string url, string pkgPath, string dirPath, string versionRange,
+                     string fileSeparator, string terminalWidth) returns error? {
+    endpoint http:Client centralEndpoint = httpEndpoint;
+
     string fullPkgPath = pkgPath;
     string destDirPath = dirPath;
     http:Request req = new;
     req.addHeader("Accept-Encoding", "identity");
 
     http:Response httpResponse = new;
-    var result = httpEndpoint -> get(untaint versionRange, message=req);
+    var result = centralEndpoint -> get(untaint versionRange, message=req);
 
     match result {
         http:Response response => httpResponse = response;
         error e => {
-            io:println(logger.formatLog("connection to the remote host failed : " + e.message));
-            return;
+            return createError("connection to the remote host failed : " + e.message);
         }
     }
 
     http:Response res = new;
     string statusCode = <string> httpResponse.statusCode;
     if (statusCode.hasPrefix("5")) {
-        io:println(logger.formatLog("remote registry failed for url :" + url));
+        return createError("remote registry failed for url :" + url);
     } else if (statusCode != "200") {
         var jsonResponse = httpResponse.getJsonPayload();
         match jsonResponse {
             json resp => {
-                io:println(logger.formatLog(resp.message.toString()));
+                if (!(statusCode == "404" && isBuild)) {
+                    return createError(resp.message.toString());
+                } else {
+                    // To ignore printing the error
+                    return createError("");
```

Comment:
shouldn't we return `()` here, in the success scenario?

Reply:
If we return () here it would considered that the package was found. But actually its not found. If we return an error as "package" not found, then it will get printed and the similar error of package not found will also get printed by the package loader.